packages/framework/src/validators/class-validator.validator.ts

```diff
@@ -0,0 +1,139 @@
+    const currentPath = `${parentPath}/${err.property}`.replace(/\/+/g, '/');
+
+    if (err.children && err.children.length > 0) {
+      // Recursively format the children
+      return formatErrors(err.children, currentPath);
+    } else {
+      // Base case: no children, return the formatted error
+      return {
+        path: currentPath,
+        message: Object.values(err.constraints || {}).join(', '),
+      };
+    }
+  });
+}
+
+export class ClassValidatorValidator implements Validator<ClassType> {
+  canHandle(schema: Schema): schema is ClassType {
+    return typeof (schema as ClassType) === 'function';
+  }
+
+  async validate<
+    T_Schema extends ClassType = ClassType,
+    T_Unvalidated = FromSchemaUnvalidated<T_Schema>,
+    T_Validated = FromSchema<T_Schema>,
+  >(data: T_Unvalidated, schema: T_Schema): Promise<ValidateResult<T_Validated>> {
+    try {
+      // eslint-disable-next-line global-require
+      require('reflect-metadata') as typeof import('reflect-metadata');
+      // eslint-disable-next-line global-require
+      const { plainToInstance, instanceToPlain } = require('class-transformer') as typeof import('class-transformer');
+      // eslint-disable-next-line global-require
+      const { validate } = require('class-validator') as typeof import('class-validator');
+
+      // Convert plain data to an instance of the schema class
+      const instance = plainToInstance(schema, data);
+
+      // Validate the instance
+      const errors = await validate(instance as object, { whitelist: true });
+
+      // if undefined, then something went wrong
+      if (!instance && !!data) throw new Error('Failed to convert data to an instance of the schema class');
+
+      if (errors.length === 0) {
+        return { success: true, data: instanceToPlain(instance) as T_Validated };
+      } else {
+        return {
+          success: false,
+          errors: formatErrors(errors),
+        };
+      }
+    } catch (error) {
+      if ((error as Error)?.message?.includes('Cannot find module')) {
+        throw new Error(
+          'Tried to use a class-validator schema in @novu/framework without `class-validator`, `class-transformer` or `reflect-metadata` installed. ' +
+            'Please install it by running `npm install class-validator class-transformer reflect-metadata`.'
+        );
+      }
+      throw error;
+    }
+  }
+
+  transformToJsonSchema(schema: ClassType): JsonSchema {
+    try {
+      // eslint-disable-next-line global-require
+      const { defaultMetadataStorage } = require('class-transformer/cjs/storage');
+      // eslint-disable-next-line global-require
+      const { getMetadataStorage } = require('class-validator') as typeof import('class-validator');
+      // eslint-disable-next-line global-require, prettier/prettier
+      const { targetConstructorToSchema, validationMetadatasToSchemas } =
+        require('class-validator-jsonschema') as typeof import('class-validator-jsonschema');
```

Comment:
Same applies here, we can use `import`.

Also, we need to replace the CJS import for class-transformer with an ESM entry point.

Reply:
does this mean cjs is still supported in the future or is it dropped entirely?